hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/strategy/LogFileSizeThresholdBasedCompactionStrategy.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.table.action.compact.strategy;
+
+import org.apache.hudi.avro.model.HoodieCompactionOperation;
+import org.apache.hudi.avro.model.HoodieCompactionPlan;
+import org.apache.hudi.config.HoodieWriteConfig;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * LogFileSizeThresholdBasedCompactionStrategy orders the compactions based on the total log files size,
+ * filters the file group which log files size is less than the threshold and limits the
+ * compactions within a configured IO bound.
+ *
+ * @see LogFileSizeBasedCompactionStrategy
+ * @see BoundedIOCompactionStrategy
+ * @see CompactionStrategy
+ */
+public class LogFileSizeThresholdBasedCompactionStrategy extends LogFileSizeBasedCompactionStrategy {
+  @Override
+  public List<HoodieCompactionOperation> orderAndFilter(HoodieWriteConfig config,
+                                                        List<HoodieCompactionOperation> operations,
+                                                        List<HoodieCompactionPlan> pendingCompactionWorkloads) {
+    long threshold = config.getCompactionLogFileSizeThreshold();
+    return super.orderAndFilter(config, operations.stream()
+        .filter(e -> e.getMetrics().getOrDefault(TOTAL_LOG_FILE_SIZE, 0d) >= threshold)
```

Comment:
can we move it to LogFileSizeBasedCompactionStrategy to avoid introducing the new strategy？

Reply:
fix